File: _food-post-service/src/food/food.js
module.exports = function buildMakeFood(){
    return function makeFood({
        username,
        food_name,
        category,
        ingredients,
        img_url,
        img_name,
        hashtags
    }){
        if(!username){
            throw new Error('Must provide username');
        }
        if(!food_name){
            throw new Error('Must provide food name');
        }
        if(!category){
            throw new Error('Must provide category');
        }
        if(!ingredients){
            throw new Error('Must provide food ingredients');
        }
        if(!img_url){
            throw new Error('Must provide image url');
        }
        if(!img_name){
            throw new Error('Must provide image name');
        }

        return Object.freeze({
            getUsername: () => username,
            getFoodName: () => food_name,
            getFoodCategory: () => category,
            getIngredients: () => ingredients,
            getImageUrl: () => img_url,
            getImageName: () => img_name,
            getHashtags: () => hashtags
        });
        
    }
}

File: clickhit/app/src/main/java/com/example/clickhit/Model/Food.java
package com.example.clickhit.Model;

import java.util.List;

public class Food {
    String imgUrl, imgName, category, foodName;
    List<String> ingredients;

    public Food(String imgUrl, String imgName, String category, String foodName, List<String> ingredients) {
        this.imgUrl = imgUrl;
        this.imgName = imgName;
        this.category = category;
        this.foodName = foodName;
        this.ingredients = ingredients;
    }

    public String getImgUrl() {
        return imgUrl;
    }

    public String getImgName() {
        return imgName;
    }

    public String getCategory() {
        return category;
    }

    public String getFoodName() {
        return foodName;
    }

    public List<String> getIngredients() {
        return ingredients;
    }
}
